### PR TITLE
Fixed TODO for non-destructive CAPS type name handling

### LINKED ISSUE #159: Renamed types on consecutive UPPERCASE chars
Hi guys, first of all, thank you for your great work!
I just noticed an unexpected behaviour with the converter.

When having type names with several uppercase chars in a row, in the result get lowercased. Only the first character remains capital.

Example:

```
{
    "id":"#",
    "definitions": {
        "exampleDTO": {
            "type": "object",
            "title": "exampleDTO",
            "properties": {
                "foo": {
                    "type": "string"
                }
            }
        }
    }
}
```
leads to
```
export interface ExampleDto {
  foo?: string;
  [k: string]: any;
}
```
the originally named ExampleDTO is generated as interface ExampleDto.

I think that should be fixed as this could also lead to naming conflicts.

### PR DESCRIPTION
**Pull request by henkesn**
As this commit changes expected output, it should be considered to be breaking.

Fixes #159

### CODE REVIEW

**Comment by henkesn** [comment]
Btw, I haven't added a separate e2e test because the uppercase issue is already covered by almost all other tests. I can add one of course if it's really required. Snapshot is updated.

**Comment by henkesn** [comment]
E2e test added.

File: test/e2e/safeTypeNames.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "type": "object",
+  "properties": {
+    "foo": {
+      'type': 'object',
+      anyOf: [
+        {$ref: "#/definitions/stra'nge#name"},
+        {$ref: "#/definitions/keepUPPERCASE"},
+        {$ref: "#/definitions/under_score"},
+        {$ref: "#/definitions/camelCase"},
+        {$ref: "#/definitions/kebab-case"}
```
Comment (bcherny) [inline test/e2e/safeTypeNames.ts]: Can you also add test cases for:

- Names that contain or start with spaces
- Names that contain or start with numbers
- Names that contain or start with periods
- Names that contain or start with commas
- Names that contain or start with *, $, or !
- Names that start with uppercase characters
- Words that are all uppercase

**Comment by henkesn** [comment]
Btw, in the e2e tests, I was not able to add more than 5 alternatives to the anyOf construct. All alternatives after where just cut off. That's why I have switched to the `unreachableDefinitions` style. Is this a known behavior or is this worth another issue?

**Comment by henkesn** [comment]
@bcherny, any news about this?

**Comment by bcherny** [comment]
Hey @henkesn, I'm just getting around to this. Do you mind rebasing, then I'll be happy to merge :)

**Comment by henkesn** [comment]
@bcherny, done.

**Comment by bcherny** [comment]
Sorry @henkesn - one more time please (you're first in the merge queue this time!)

**Comment by henkesn** [comment]
@bcherny here you are.

**Pull request merged**